### PR TITLE
Basic UI

### PR DESCRIPTION
**Pull request by rugwizangoga**
# OOP school library: add basic UI

- Created app.rb file that will serve as the console app entry-point. It has methods that do the following:
                    List all books.
                    List all people.
                    Create a person (teacher or student, not a plain Person).
                    Create a book.
                    Create a rental.
                    List all rentals for a given person id.
-  In the main.rb defined there is an entry point, this will be a method called main that is invoked at the end of the file.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    puts 'Age:'
+    age = gets.chomp
+    puts 'Has parent permission? [y/n]:'
+    permission = gets.chomp.downcase
+    parent_permission = permission == 'y'
+    puts 'Classroom:'
+    classroom = gets.chomp
+    @people.push Student.new(age, classroom, parent_permission: parent_permission, name: name)
+    puts 'Person created successfully'
+  end
```
Comment (KanzaTahreem) [inline app.rb]: - [ ] Good job for creating a student 👍👍 However, I noticed that the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) does not mention anything about the student's classroom. Please make sure to follow the [project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_ui.md) and fix this ✨⛏

  image

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    date = gets.chomp
+
+    if renter.can_use_services?
+      @rentals.push(Rental.new(date, rented_book, renter))
+      puts 'Rental created successfully'
+    else
+      puts 'The person lacks borrowing permissions'
+    end
+  end
+
```
Comment (KanzaTahreem) [inline app.rb]: - [ ] Good job for creating a rental 👍👍  However, I encountered an issue where the app crashes when attempting to create a rental for a student. ⚠ Could you please address this problem and ensure that the app operates smoothly for a good user experience? ✨

   image

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @rugwizangoga 

Good job so far! I really appreciate the effort you put in!  **_However, there are few this that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Good job with the UI
- ✅ Descriptive PR title and description
- ✅ Correct Linters

## Required Changes 🛑

_Please check the comments under the review._

## Optional suggestions ❕


N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @rugwizangoga, 

To highlight 👍🏽 
✔️ App runs smoothly
✔️ Design matches wireframe

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**